### PR TITLE
feat: implement pagination in migrations table

### PR DESCRIPTION
**Pull request by goga-m**
## Summary
Closes https://app.clickup.com/t/86775vn2q

## Checklist

-   [x] My changes look good in both light AND dark mode
-   [x] The change is not hardcoded to a single network, but has multi-asset in mind
-   [x] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

**Review by samharperpittam — CHANGES_REQUESTED** [review]
Comments left on Slack in relation to:
- View More button sometimes not populating additional transactions
- View state persists resulting in a useless "View More" button

**Comment by goga-m** [comment]
@samharperpittam the pagination should be persistent now, and also adjusted the sorting of the migration transactions list so that it wouldn't change/hide transactions list when you cilck "view more". Let me know if you still see issues. Ty!

**Comment by ItsANameToo** [comment]
conflicts @goga-m

**Comment by goga-m** [comment]
The functionality is affected after the merge conflicts. Will move this to draft and re-open once it's working

**Review by samharperpittam — DISMISSED** [review]

File: src/domains/migration/hooks/use-migration-transactions.ts [hunk 1]
```diff
@@ -12,60 +12,80 @@ import {
...
+};
+
 export const useMigrationTransactions = ({ profile }: { profile: Contracts.IProfile }) => {
 	const { profileIsRestoring } = useConfiguration();
 	const { migrations, storeTransactions } = useMigrations();
 	const [latestTransactions, setLatestTransactions] = useState<DTO.ExtendedConfirmedTransactionData[]>([]);
 	const [isLoadingTransactions, setIsLoadingTransactions] = useState(true);
+	const [hasMore, setHasMore] = useState(false);
+	const [page, setPage] = useState(0);
+	const limit = 10;
```
Comment (ItsANameToo) [inline src/domains/migration/hooks/use-migration-transactions.ts:71]: this limit needs to be applied to the API call, now you slice an array with 100 values and leave 90 out from the UI and when paginating you request another 100, etc

**Comment by ItsANameToo** [comment]
failing test issue + pagination issue, see slack

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**